lib/cog/relay/relays.ex

```diff
@@ -96,38 +94,10 @@ defmodule Cog.Relay.Relays do
 
   ########################################################################
 
-  # Common processing of relay announcements, whether they come from a
-  # real Relay instance, or from the bot itself, announcing the
-  # embedded bundle.
-  defp process_discovery(discovery, state) when is_map(state) do
-    case discovery do
-      %{"intro" => intro} ->
-        process_introduction(intro, state)
-      %{"announce" => announcement} ->
-        process_announcement(announcement, state)
-    end
-  end
-
-  defp process_introduction(%{"relay" => id,
-                              "reply_to" => reply_to}, state) do
-    case CredentialManager.get(id, by: :id) do
-      {:ok, nil} ->
-        creds = %Credentials{id: id}
-        CredentialManager.store(creds)
-        Logger.info("Stored credentials for Relay #{id}")
-      {:ok, _} ->
-        :ok
-    end
-    {:ok, my_creds} = CredentialManager.get()
-    Messaging.Connection.publish(state.mq_conn, %{intro: %{id: my_creds.id,
-                                                           role: "bot"}}, routed_by: reply_to)
-    state
-  end
-
-  defp process_announcement(announcement, %__MODULE__{tracker: tracker}=state) do
+  defp process_announcement(announcement, %__MODULE__{tracker: tracker}=state, internal \\ false) do
```

Comment:
Could `internal` be an atom instead of a boolean? Makes for easier-to-read functions, IMO.